src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/CourseSchedule.kt

```diff
@@ -0,0 +1,52 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.builders
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseActivity
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseSchedule
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.SlotCategory
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.CourseScheduleRepository
+import java.time.LocalDate
+import java.time.LocalTime
+
+@DslMarker
+annotation class CourseScheduleDslMarker
+
+@TestDataDslMarker
+interface CourseScheduleDsl
+
+@Component
+class CourseScheduleBuilderRepository(val courseScheduleRepository: CourseScheduleRepository) {
+  fun save(courseSchedule: CourseSchedule) = courseScheduleRepository.save(courseSchedule)
+}
+
+@Component
+class CourseScheduleBuilderFactory(val repository: CourseScheduleBuilderRepository? = null) {
+  fun builder() = CourseScheduleBuilder(repository)
+}
+
+class CourseScheduleBuilder(val repository: CourseScheduleBuilderRepository? = null) : CourseScheduleDsl {
+  fun build(
+    courseActivity: CourseActivity,
+    courseScheduleId: Long,
+    scheduleDate: String,
+    startTime: String,
+    endTime: String,
+    slotCategory: SlotCategory,
+    scheduleStatus: String,
+  ) =
+    LocalDate.parse(scheduleDate).let { date ->
+      CourseSchedule(
+        courseScheduleId = courseScheduleId,
+        courseActivity = courseActivity,
+        scheduleDate = date,
+        startTime = date.atTime(LocalTime.parse(startTime)),
+        endTime = date.atTime(LocalTime.parse(endTime)),
+        slotCategory = slotCategory,
+        scheduleStatus = scheduleStatus,
+      ).let {
+        save(it)
+      }
+    }
```

Comment:
When we build the CourseSchedule we just save and return so it can be assigned in the test.